egs2/librispeech/asr1/conf/tuning/train_asr_s4_decoder.yaml

```diff
@@ -0,0 +1,96 @@
+# Trained with Tesla A100 (40GB) x 4 GPUs. It takes about 2.5 days.
+encoder: conformer
+encoder_conf:
+    output_size: 512
+    attention_heads: 8
+    linear_units: 2048
+    num_blocks: 12
+    dropout_rate: 0.1
+    positional_dropout_rate: 0.1
+    attention_dropout_rate: 0.1
+    input_layer: conv2d
+    normalize_before: true
+    macaron_style: true
+    rel_pos_type: latest
+    pos_enc_layer_type: rel_pos
+    selfattention_layer_type: rel_selfattn
+    activation_type: swish
+    use_cnn_module: true
+    cnn_module_kernel: 31
+
+decoder: s4
+decoder_conf:
+  dropinp: 0.0
+  dropout: 0.1
+  drop_path: 0.1
+  prenorm: true
+  n_layers: 6
+  # specify each block config here. for detail, see espnet2/asr/state_spaces 
+  # layer: stack of black box modules, _name_ must be specified
+  # residual: choose an option for residual connection
+  layer:
+    - _name_: s4
+      postact: glu
+      dropout: 0.1
+      n_ssm: 1
+      lr: 0.0025
+      dt_min: 0.001
+      dt_max: 0.1
+      measure: legs
+    - _name_: mha
+      n_head: 8
+      dropout: 0.1
+    - _name_: ff
+      expand: 4
+      activation: gelu
+      dropout: 0.1
+  residual: residual
+  norm: layer
+
+model_conf:
+    ctc_weight: 0.3
+    lsm_weight: 0.1
+    length_normalized_loss: false
+
+frontend_conf:
+    n_fft: 512
+    hop_length: 160
+
+use_amp: true
+num_workers: 4
+batch_type: numel
+batch_bins: 35000000
+accum_grad: 4
+max_epoch: 60
+patience: none
+init: none
+best_model_criterion:
+-   - valid
+    - acc
+    - max
+keep_nbest_models: 10
+
+optim: adamw
+exclude_weight_decay: true
```

Comment:
Just curious: Is AdamW better than Adam? Also, does excluding weight decay for bias etc improve the performance?

Reply:
In our experiments, in S4 decoder case, we confirmed that AdamW is slightly better than Adam optimizer.
While in Transformer decoder case, there was no significant difference. 
For the weight decay trick, we followed the experimental setup of S4. Perhaps the effect may not be so apparent.

- S4 results (w/o LM)

|                     | dev_clean | dev_other | test_clean | test_other |
|---------------------|-----------|-----------|------------|------------|
| Adam                | 2.1       | 5.1       | 2.2        | 5.2        |
| AdamW               | 2.0       | 5.0       | 2.3        | 5.0        |
| AdamW w/o excluding | 2.0       | 5.0       | 2.2        | 5.0        |

Reply:
Thanks!

Reply:
Kind of out of scope but did someone experiment with the `ScaledAdam` from k2 team?